File: mc_orbits/fornax/sample.py
import numpy as np
from lilguys import coords, Snapshot, units

fornax_obs = coords.observation(
    ra = 39.9970833,
    dec = -34.4491667,
    distance = 147,
    distance_err = 12,
    pm_ra = 0.374 ,
    pm_ra_err = 0.035 ,
    pm_dec = -0.401,
    pm_dec_err = 0.035,
    radial_velocity = 55.3,
    radial_velocity_err = 0.3
)


N = 10_000

mc_obs = coords.rand_coords(fornax_obs, N)
mc_phase = [coords.to_galcen(o) for o in mc_obs]
pos = np.array([[p.x, p.y, p.z] for p in mc_phase])
vel = np.array([[p.v_x, p.v_y, p.v_z] for p in mc_phase])
vel /= units.V_0
vel *= -1
pos /= units.R_0
snap = Snapshot(pos, vel, m=0)

snap.save("positions.hdf5")

File: models/sculptor/mc_hq/sample.jl
using LilGuys


const obs = LilGuys.Observation(
    ra = 15.03917,
    dec = 33.70917,
    distance = 100, #86,
    pm_ra = 0.099,
    pm_dec = -0.160,
    radial_velocity = 111.4,
)

const err = LilGuys.Observation(
    ra = 0.1,
    dec = 0.1,
    distance = 3,
    pm_ra = 0.02,
    pm_dec = 0.02,
    radial_velocity = 10,
)


function sample(N = 1)
    mc_obs = LilGuys.rand_coords(obs, err, N)
    pushfirst!(mc_obs, obs) # add the true observation to the sample

    mc_phase = [LilGuys.to_galcen(o) for o in mc_obs]
    pos = reduce(hcat, [[p.position...] for p in mc_phase])
    vel = reduce(hcat, [[p.velocity...] for p in mc_phase])

    vel *= -1 # reverse velocities to go backwards in time
    m = 1e-20
    snap = Snapshot(pos, vel, LilGuys.ConstVector(m, N+1))
    return snap
end


if abspath(PROGRAM_FILE) == @__FILE__
    snap = sample()
    save("positions.hdf5", snap)
end
